src/Framework/test/PackageTests.cs

```diff
@@ -0,0 +1,143 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO.Compression;
+using System.Linq;
+using System.Reflection.Metadata;
+using System.Reflection.PortableExecutable;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Testing;
+using Xunit.Abstractions;
+
+namespace Microsoft.AspNetCore;
+
+public class PackageTests
+{
+    private readonly string _packageLayoutRoot;
+    private readonly ITestOutputHelper _output;
+
+    public PackageTests(ITestOutputHelper output)
+    {
+        // In official builds, only run this test on Helix. The test will work locally so long as we're building w/ shipping versions.
+        if (!TestData.VerifyPackageAssemblyVersions() && !SkipOnHelixAttribute.OnHelix())
+        {
+            return;
+        }
+
+        _output = output;
+        _packageLayoutRoot = SkipOnHelixAttribute.OnHelix() ?
+            Path.Combine(
+                Environment.GetEnvironmentVariable("DOTNET_ROOT"),
+                "Packages.Layout") :
+            TestData.GetPackageLayoutRoot();
+        var packageRoot = SkipOnHelixAttribute.OnHelix() ?
+            Environment.GetEnvironmentVariable("HELIX_WORKITEM_ROOT") :
+            TestData.GetPackagesFolder();
+        var packages = Directory
+                        .GetFiles(packageRoot, "*.nupkg", SearchOption.AllDirectories)
+                        .Where(file => !file.EndsWith(".symbols.nupkg", StringComparison.OrdinalIgnoreCase));
+
+        if (Directory.Exists(_packageLayoutRoot))
+        {
+            Directory.Delete(_packageLayoutRoot, true);
+        }
+
+        foreach (var package in packages)
+        {
+            var outputPath = Path.Combine(_packageLayoutRoot, Path.GetFileNameWithoutExtension(package));
+            ZipFile.ExtractToDirectory(package, outputPath);
+        }
+    }
+
+    [Fact]
+    public void PackageAssembliesHaveExpectedAssemblyVersions()
+    {
+        // In official builds, only run this test on Helix. The test will work locally so long as we're building w/ shipping versions.
+        if (!TestData.VerifyPackageAssemblyVersions() && !SkipOnHelixAttribute.OnHelix())
```

Comment:
The above comment doesn't need to be extended, given the information is visible not too far above. However, suggest
```suggestion
        if (!TestData.VerifyPackageAssemblyVersions() && !SkipOnCIAttribute.OnCI())
```